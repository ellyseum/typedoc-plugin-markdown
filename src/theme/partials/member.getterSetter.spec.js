describe(`[partial] member.gettersetter`, () => {
  test('should compile getter', () => {
    const context = {
      name: 'readOnlyNameProperty',
      originalName: 'readOnlyNameProperty',
      getSignature: {},
    };
    const result = global.compileTemplate('member.getterSetter.hbs', context);
    expect(result).toMatchSnapshot();
  });

  test('should compile setter', () => {
    const context = {
      setSignature: {
        name: '__set',
        flags: [],
        id: 91,
        originalName: '__set',
        kind: 1048576,
        parameters: [
          {
            name: 'value',
            flags: [],
            id: 92,
            parent: null,
            originalName: 'value',
            kind: 32768,
            type: { type: 'intrinsic', name: 'string' },
          },
        ],
        type: { type: 'intrinsic', name: 'void' },
        kindString: 'Set signature',
      },
    };
    const result = global.compileTemplate('member.getterSetter.hbs', context);
    expect(result).toMatchSnapshot();
  });
});
