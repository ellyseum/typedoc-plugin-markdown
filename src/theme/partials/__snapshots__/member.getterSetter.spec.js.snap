// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[partial] member.gettersetter should compile getter 1`] = `
"get readOnlyNameProperty [member.signature.title]

[member.signature.body]"
`;

exports[`[partial] member.gettersetter should compile setter 1`] = `
"set  [member.signature.title]

[member.signature.body]"
`;
